### PR TITLE
optimize queries to mongo to reduce load on flask

### PR DESCRIPTION
**Pull request by christopherjwang**

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -15,29 +15,40 @@ def index():
...
+    result = list(courses.aggregate(aggregationPipeline))
+    classes = result[0].get('classes') if len(result) > 0 else None
+    
     return jsonify({'numbers': classes})
 
 @app.route('/spring2016/<school>/<number>')
 def single_course(school, number):
-    course = courses.find_one({'school':school, 'number':number})
-    del course['_id']
+    course = courses.find_one({'school':school, 'number':number}, {'_id': 0})
```
Comment (joshuamorton) [inline app.py:46]: Will this delete the attribute when returned, or simply set it to 0 when sent to the user?

Comment (christopherjwang) [inline app.py:46]: Delete. It actually just doesn't return it. It's called projection if you look at docs for find here: http://api.mongodb.org/python/current/api/pymongo/collection.html

**Comment by joshuamorton** [comment]
Sweet! Thanks, I'll merge this shortly

**Comment by joshuamorton** [comment]
And its in production and working.

**Comment by christopherjwang** [comment]
Sweet. Happy holidays!

**Pull request merged**